frontend/util/useQueryParameter.tsx

```diff
@@ -11,5 +11,5 @@ export function useQueryParameter(parameter: string) {
   if (typeof checkingParameter === "string") {
     return checkingParameter
   }
-  return checkingParameter[-1]
+  return checkingParameter.slice(-1)[0]
```

Comment:
checkingParameter[checkingParameter.length - 1] would do the same thing without an extra allocation.